app/controllers/tags_controller.rb

```diff
@@ -94,6 +94,26 @@ def select_merge; end
 
   def merge
     @primary = @tag
+
+    # No merge to self.
+    if params[:merge_with_id].eql? @primary.id.to_s
+      flash[:danger] = 'Cannot merge a tag with itself.'
+
+      # redirect_to tag_path(id: @category.id, tag_id: @primary.id) && return
+
+      # render :edit, status: 400
```

Comment:
This worked better than the previous one, but didn't do what anyone would expect it the page to do, TBH I was seeing it coming.